tools/sourcemap-tools/src/SourceProcessor.ts

```diff
@@ -0,0 +1,71 @@
+import fs from 'fs';
+import { BasicSourceMapConsumer, Position, RawSourceMap, SourceMapConsumer, SourceMapGenerator } from 'source-map';
+import { DebugIdGenerator } from './DebugIdGenerator';
+import { stringToUuid } from './helpers/stringToUuid';
+
+export class SourceProcessor {
+    constructor(private readonly _debugIdGenerator: DebugIdGenerator) {}
+
+    public async processSourceAndSourceMap(source: string, sourceMap: string | RawSourceMap, debugId?: string) {
```

Comment:
what is source? Is this the path to source map? Is this the source map content? I checked the code however it makes sense to make leave the note

Reply:
`source` is the source file content. `sourceMap` is the source map JSON or object.

Will add some docs.